react/features/base/profile/functions.js

```diff
@@ -5,62 +5,69 @@ import { toState } from '../redux';
 
 /**
  * Returns the effective value of a property by applying a precedence
- * between values in URL, config and profile.
+ * between values in JWT, URL, config and profile.
  *
  * @param {Object|Function} stateful - The redux state object or function
  * to retreive the state.
  * @param {string} propertyName - The name of the property we need.
  * @param {{
- *     ignoreJWT: boolean,
- *     ignoreUrlParams: boolean,
- *     ignoreProfile: boolean,
- *     ignoreConfig: boolean
- * }} precedence - A structure of booleans to set which property sources
- * should be ignored.
+ *     config: boolean,
+ *     jwt: boolean,
+ *     profile: boolean,
+ *     urlParams: boolean
+ * }} sources - A structure of booleans to set which property sources to be
+ * included.
  * @returns {any}
  */
 export function getPropertyValue(
         stateful: Object | Function,
         propertyName: string,
-        precedence: Object = {
-            ignoreJWT: false,
-            ignoreUrlParams: false,
-            ignoreProfile: false,
-            ignoreConfig: false
-        }
+        sources: Object
 ) {
+    // Default values don't play nicely with partial objects and we want to make
+    // the function to be easily usable without defining every fields,
+    // defaulting to true, e.g.
+    // getPropertyValue(state, name, {urlParams: false})
+    const _sources = {
+        config: true,
+        jwt: true,
+        profile: true,
+        urlParams: true,
+        ...sources
+    };
     const state = toState(stateful);
-    const jwt = state['features/base/jwt'];
-    const urlParams
-        = parseURLParams(state['features/base/connection'].locationURL);
-    const profile = state['features/base/profile'];
-    const config = state['features/base/config'];
-    const urlParamName = `config.${propertyName}`;
 
     // Precedence: jwt -> urlParams -> profile -> config
 
-    if (
-        !precedence.ignoreJWT
-        && typeof jwt[propertyName] !== 'undefined'
-    ) {
-        return jwt[propertyName];
+    if (_sources.jwt) {
+        const jwt = state['features/base/jwt'];
+
+        if (typeof jwt[propertyName] !== 'undefined') {
+            return jwt[propertyName];
+        }
     }
 
-    if (
-        !precedence.ignoreUrlParams
-        && typeof urlParams[urlParamName] !== 'undefined'
-    ) {
-        return urlParams[urlParamName];
+    if (_sources.urlParams) {
+        const urlParams
+            = parseURLParams(state['features/base/connection'].locationURL);
+        const urlParamName = `config.${propertyName}`;
+
+        if (typeof urlParams[urlParamName] !== 'undefined') {
+            return urlParams[urlParamName];
```

Comment:
My example source code explicitly avoided the double `urlParams[urlParamName]`.

Reply:
I didn't get that, sorry, fixed. I hope it's not too much that I combined the entire statement(s)